docs/_posts/2021-07-05-whats-next.md

```diff
@@ -0,0 +1,176 @@
+
+A potential downside to this approach is that the `BrowserStore` and related `BrowserAction`s become the new _“interface”_ that a browser engine has to dispatch correctly, which can be harder to understand and follow than simply implementing actual interfaces.
+
+Overall this seems to be worthwhile exploring and potentially discussing further in an RFC.
+
+## Jetpack Compose
+
+Android’s new UI toolkit, [Jetpack Compose](https://developer.android.com/jetpack/compose), will significantly change how we build user-facing features in components and apps.
+
+The good news is that Jetpack Compose works very nicely with our browser-state component. In Android Components we will provide bindings that allow subscribing to any `lib-state` baked `Store`, causing a recomposition if the observed state changes.
+
+```kotlin
+@Composable
+fun SimpleToolbar(
+    store: BrowserStore
+) {
+    // Subscribe to the URL of the selected tab
+    val url = store.observeAsState { state -> state.selectedTab?.content?.url }
+
+    // Will automatically get recomposed if the URL changes
+    Text(url.value ?: "")
+}
+```
+
+### Observing specific tabs
+
+Today we have many components that optionally take a nullable `tabId: String?` parameter. If a tab ID is provided then the component is supposed to observe this specific tab. And if the parameter is `null` then the component will automatically track the currently selected tab. This has caused issues in the past when a `null` value was provided accidentally, causing the wrong tab to be tracked. With Jetpack Compose we want to make this more explicit and will provide a `Target` class, that lets the caller explicily define what tab should be targeted. In addition to that this allows us to provide extension functions for easily observing this tab.
+
+```kotlin
+@Composable
+fun Example(store: BrowserStore) {
+    // Explicitly observe specific tabs
+    SimpleToolbar(store, Target.SelectedTab)
+    SimpleToolbar(store, Target.Tab("tabId"))
+    SimpleToolbar(store, Target.CustomTab("customTabId"))
+}
+
+@Composable
+fun SimpleToolbar(
+    store: BrowserStore,
+    target: Target
+) {
+    // Observe the URL of the target. Only when the URL changes, this will
+    // cause a recomposition. Other changes of the tab get ignored.
+    val tab: SessionState? by target.observeAsStateFrom(
+        store = store,
+        observe = { tab -> tab?.content?.url }
+    )
+
+    Text(tab?.content?.url ?: "")
+}
+```
+
+## Observing state directly vs concept components
+
+A central piece of our [current component architecture](/contributing/architecture) is splitting the implementation into three pieces: A concept component, an implementation, and a glue/feature component for integration. This allows us to easily swap (concept) implementations, without having to change any other code. The downside of this approach is that all implementations need to abide by the interface abstractions.
```

Comment:
I am not sure this is a downside :). We got a lot of pluggability out of it, and I find it easy to wire components together based on their static (interface) contract. I am afraid this will become much harder to consume indepdently if it's just a loose set of actions that define the contract, but would love to prototype so we can make up our mind. I does appear to make things easier for us knowing the system well.